bpaf_core/src/visitor/explain_unparsed.rs

```diff
@@ -127,6 +130,25 @@ impl<'a> ExplainUnparsed<'a> {
         }
         None
     }
+    fn is_redundant(&self, parsed: &[Name], unparsed: Name) -> Option<Error> {
+        let unparsed_info = self.all_names.get(&unparsed)?;
+        if unparsed_info.in_many {
+            return None;
+        }
+        for p in parsed {
+            let Some(parsed) = self.all_names.get(p) else {
+                continue;
+            };
+            if !parsed.in_many {
```

Comment:
I don't think this check is correct. `parsed` is unrelated to `unparsed`...

Reply:
I assume that your note meant that.

Let see if it can be made simpler